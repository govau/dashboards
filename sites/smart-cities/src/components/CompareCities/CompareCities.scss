@import '~style/main';

.container {
  @include container;
}

.wrapper {
  @include shadow;

  margin: 32px 0;
  border-radius: 8px;
  background: $color-white;
  border: 1px solid $color-grey--200;
}

.header {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 16px;
  border-bottom: 1px solid $color-grey--200;

  @include for-tablet-portrait-up {
    flex-direction: row;
    align-items: baseline;
    padding: 24px 30px;
  }
}

.title {
  margin-bottom: 8px;
  font-size: 24px;
  font-weight: 300;
  line-height: 1;

  @include for-tablet-portrait-up {
    flex-direction: row;
    margin-right: 1ch;
    margin-bottom: 0;
  }
}

.subTitle {
  font-size: 20px;
  font-weight: 500;
  line-height: 1.4;

  @include for-tablet-landscape-up {
    font-size: 22px;
    font-weight: 300;
  }
}

.body {
  padding: 15px;
  columns: 140px;
  column-gap: 0;
  background: $color-grey--060;

  @include for-tablet-landscape-up {
    padding: 31px;
  }
}

.checkbox {
  display: block;
  break-inside: avoid-column;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  margin-bottom: 13px;
  font-size: 12px;
}
