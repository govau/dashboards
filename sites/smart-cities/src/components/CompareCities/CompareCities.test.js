import React from 'react';
import { shallow } from 'enzyme';
import CompareCities from './CompareCities';

jest.mock('../Checkbox/Checkbox', () => 'Checkbox');

const toggleCitySelectedMock = jest.fn();

const defaultProps = {
  city: {
    id: 'perth',
    name: 'Greater Perth',
  },
  cities: [
    {
      id: 'perth',
      name: 'Perth',
    },
    {
      id: 'sydney',
      name: 'Sydney',
    },
    {
      id: 'melbourne',
      name: 'Melbourne',
    },
  ],
  toggleCitySelected: toggleCitySelectedMock,
  category: {
    colorName: 'jobs',
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CompareCities {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render a checkbox for each city except the current one', () => {
  const component = shallow(
    <CompareCities {...defaultProps} />
  );

  expect(component.find('Checkbox').length).toBe(defaultProps.cities.length - 1);
});

it('should base the checkbox color on the category color', () => {
  const component = shallow(
    <CompareCities {...defaultProps} />
  );

  const firstCheckbox = component.find('Checkbox').at(0);

  expect(firstCheckbox.prop('checkedBoxColor')).toBe('JOBS_500');
});

it('should show the count of other cities in the title', () => {
  const component = shallow(
    <CompareCities {...defaultProps} />
  );

  const subTitle = component.find('.subTitle');

  expect(subTitle.text()).toBe(' across 2 other cities');
});
