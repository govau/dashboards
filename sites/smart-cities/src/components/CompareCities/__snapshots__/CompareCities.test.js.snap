// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"container\\">
  <div className=\\"wrapper\\">
    <div className=\\"header\\">
      <h2 className=\\"title\\">
        Compare 
        Greater Perth
      </h2>
      <span className=\\"subTitle\\">
         across 
        2
         other cities
      </span>
    </div>
    <div className=\\"body\\">
      <Checkbox checked={[undefined]} checkedBoxColor=\\"JOBS_500\\" className=\\"checkbox\\" id=\\"sydney\\" label=\\"Sydney\\" onChange={[Function]} />
      <Checkbox checked={[undefined]} checkedBoxColor=\\"JOBS_500\\" className=\\"checkbox\\" id=\\"melbourne\\" label=\\"Melbourne\\" onChange={[Function]} />
    </div>
  </div>
</div>"
`;
